### ISSUE TITLE
App crashed when back from motionView to mainView

### ISSUE THREAD

**Issue by peace-v** [iSame7/Panoramic#2]
My app's mainView is a tableViewController, click the cell, it will push to the motionView, but when hit the back button, app will crash and no message be printed in console, the breakpoint just points to the line `class AppDelegate: UIResponder, UIApplicationDelegate {`. And when I comment the code below

<pre><code>let motionView = PanoramaView(frame: self.view.bounds)
motionView.setImage(UIImage(named: "space")!)
self.view.addSubview(motionView)</code></pre>

everything works fine. And I also tried to present modally, still the same result.

**Comment by msetten** [comment]
I am experiencing similar issues. Only not every time when I get back from my apps main tableViewController. Sometimes I can even go to another panoramic image, but other times it crashes without any messages when loading a second image. @peace-v Did you every find a solution for this?

**Comment by iSame7** [comment]
I looked into this issue before but was not able to reproduce at my end. let me try again. If you are integrating crash reporting tool in your app and you can send me the crash report that would helpful.

**Comment by peace-v** [comment]
@msetten I didn't figure out, so just used another framework to work it out which called SubjectiveCPhotoPanner. [https://github.com/subjc/SubjectiveCPhotoPanner](url)

**Comment by peace-v** [comment]
@iSame7 Thanks for your reply. But it has been some time, I'm afraid that I cannot find the crash report at that time.

**Comment by iSame7** [comment]
@peace-v @msetten No problem i'll try to reproduce and fix this issue as soon as possible, In the mean time you could take a look at https://github.com/iSame7/Panorama the same control but in Objective C I've developed it before i write this Swift version of it.

**Comment by msetten** [comment]
I believe I have found the issue, which is not in your code. I used full resolution panoramic photos that are taken on the iPhone. This increased the memory usage so much that at some point the app was the app with the most memory in use and thus the first to be automatically stopped by iOS when memory got too low. When I now request the photos framework for a resized version that fits the device's screen height (and width adjusted by ratio) the crashes stop.
